torch/_functorch/aot_autograd.py

```diff
@@ -423,9 +423,90 @@
 aot_autograd_decompositions = {}
 
 
+FakifiedFlatArgs = NewType("FakifiedFlatArgs", List[Any])
+
+
+def process_inputs(
+    flat_args: List[Any],
+    aot_config: AOTConfig,
+    fake_mode: FakeTensorMode,
+    shape_env: Optional[ShapeEnv],
+) -> FakifiedFlatArgs:
+    def convert(idx, x):
+        if shape_env is not None:
+            from torch._dynamo.source import ConstantSource
+
+            if isinstance(x, int):
+                # We always specialize on scalar values in export.
+                if aot_config.is_export:
+                    return x
+                source = ConstantSource(f"sym_{idx}")
+                return shape_env.create_symintnode(
+                    shape_env.create_symbol(x, source), hint=x, source=source
+                )
+        if isinstance(x, torch.ScriptObject):
+            return torch._library.fake_class_registry.maybe_to_fake_obj(fake_mode, x)
+        if not isinstance(x, torch.Tensor):
+            return x
+        if isinstance(x, FakeTensor):
+            assert x.fake_mode is fake_mode
+            return x
+        if is_traceable_wrapper_subclass(x):
+            attrs, _ = x.__tensor_flatten__()
+            if all(isinstance(getattr(x, attr), FakeTensor) for attr in attrs):
+                assert all(getattr(x, attr).fake_mode is fake_mode for attr in attrs)
```

Comment:
cc @tugsbayasgalan this seems like a spot where we missed handling nested wrapper subclasses today? (I'm actually not sure why this doesn't obviously blow up our nested subclass tests, since we are asserting here that the inner tensors are all FakeTensors)

Reply:
Hmm yeah does it mean that it will just create fresh fake tensors? This probably means nested subclasses don't work with dynamic shapes today? Because these fake tensors would be coming from dynamo i think

Reply:
I will make a patch after this PR lands

Reply:
> This probably means nested subclasses don't work with dynamic shapes today

i might be missing something, but it seems to me like this code also runs when there are static shapes?

Reply:
Oh what i meant is it today probably just ignores what dynamo gives aotdispatcher and creates fresh fake tensors as a result any new information added from dynamo is ignored